### PR TITLE
Add import course from zip file

### LINKED ISSUE #6: Import course from ZIP

### PR DESCRIPTION
**Pull request by mako321**
close #6

### CODE REVIEW

File: src/Services/ExportImportService.php [hunk 1]
```diff
@@ -85,8 +106,118 @@ public function export($courseId): string
...
+
+        return $lesson;
+    }
+
+    private function createTopicFromImport(array $topicData, string $dirFullPath): Model
+    {
+        $topicData = array_merge($topicData, $topicData['topicable'] ?? []);
+        unset($topicData['topicable']);
+
+        dd(TopicFacade::availableContentClasses());
```
Comment (mako321) [inline src/Services/ExportImportService.php]: 'topicable_type' => ['string', Rule::in(TopicFacade::availableContentClasses())],

when I run the tests it returns empty []

when I run by postman array:7 [
  0 => "EscolaLms\TopicTypes\Models\TopicContent\Audio"
  1 => "EscolaLms\TopicTypes\Models\TopicContent\Video"
  2 => "EscolaLms\TopicTypes\Models\TopicContent\Image"
  3 => "EscolaLms\TopicTypes\Models\TopicContent\RichText"
  4 => "EscolaLms\TopicTypes\Models\TopicContent\H5P"
  5 => "EscolaLms\TopicTypes\Models\TopicContent\OEmbed"
  6 => "EscolaLms\TopicTypes\Models\TopicContent\PDF"
]

Comment (qunabu) [reply src/Services/ExportImportService.php]: you need to add [https://github.com/EscolaLMS/topic-types/blob/main/src/EscolaLmsTopicTypesServiceProvider.php](EscolaLmsTopicTypesServiceProvider) to testbench service providers https://github.com/EscolaLMS/Courses-Import-Export/blob/main/tests/TestCase.php#L42

you can read more about this https://packages.tools/testbench/

File: src/Services/ExportImportService.php [hunk 2]
```diff
@@ -85,8 +105,116 @@ public function export($courseId): string
...
+        $dirPath = 'imports' . DIRECTORY_SEPARATOR . 'courses' . DIRECTORY_SEPARATOR . uniqid(rand(), true);
+        $dirFullPath = $this->extractZipFile($zipFile, $dirPath);
+        $content = json_decode(File::get($dirFullPath . DIRECTORY_SEPARATOR . 'content.json'), true);
+        $course = DB::transaction(function () use ($content, $dirFullPath) {
+            return $this->createCourseFromImport($content, $dirFullPath);
+        });
+
+        Storage::deleteDirectory($dirPath);
+
+        return $course;
```
Comment (qunabu) [inline src/Services/ExportImportService.php]:
```suggestion
        $dirPath = 'imports' . DIRECTORY_SEPARATOR . 'courses' . DIRECTORY_SEPARATOR . uniqid(rand(), true);
        $dirFullPath = $this->extractZipFile($zipFile, $dirPath);
        
        try { 
        
        $content = json_decode(File::get($dirFullPath . DIRECTORY_SEPARATOR . 'content.json'), true);
        $course = DB::transaction(function () use ($content, $dirFullPath) {
            return $this->createCourseFromImport($content, $dirFullPath);
        });
        
        } catch (Exception $e) {
         // log error or nothing 
        } finally { 
            Storage::deleteDirectory($dirPath);
        }

       

        return $course;
```

**Review by qunabu — CHANGES_REQUESTED** [review]
make sure 
- admin user gets and error is import fails (add test for failure with random zip)
- always clean after attempt, delete folder even when import failed

**Review by qunabu — APPROVED** [review]

**Pull request merged**